scripts/setup.sh

```diff
@@ -1,18 +1,11 @@
-#!/usr/bin/env bash
+#!/bin/sh
 set -e
 
-if ! command -v pdm &>/dev/null; then
-    if ! command -v pipx &>/dev/null; then
-        python3 -m pip install --user pipx
-    fi
-    pipx install pdm
-fi
-if ! pdm self list 2>/dev/null | grep -q pdm-multirun; then
-    pdm install --plugins
-fi
+command -v pipx &>/dev/null || python3 -m pip install --user pipx
+command -v pdm &>/dev/null || pipx install pdm
 
-if [ -n "${PDM_MULTIRUN_VERSIONS}" ]; then
-    pdm multirun -v pdm install -G:all
-else
-    pdm install -G:all
-fi
+pdm self list 2>/dev/null | grep -q pdm-multirun || pdm install --plugins
+
+[ -n "${PDM_MULTIRUN_VERSIONS}" ] && pdm multirun -v pdm install -G:all && exit 0
+
+pdm install -G:all
```

Comment:
I'd say that's one of the reason I find if-else more robust: if `pdm multirun` fails, we don't `exit 0` and therefore `pdm install` will run too, while it shouldn't. It then requires wrapping some of the conditions in `{}`:

```
{ [ -n "${PDM_MULTIRUN_VERSIONS}" ] && pdm multirun -v pdm install -G:all } || pdm install -G:all
```

That either makes long lines, or less readable lines. WDYT?

Reply:
You're right, this is a small oversight of mine.
I'd actually pad the line below with the inverse condition (`-z`).
This is one place where an if-else construct also makes sense, so if my following commit isn't to your taste we can totally have an if-else for that part

Reply:
Ah, no, because `set -e`. Yeah. But see, that makes things a bit harder to follow.

Reply:
> Ah, no, because `set -e`. Yeah. But see, that makes things a bit harder to follow.

Definitely. I don't usually like relying on `set -e` for that reason

Reply:
I myself prefer to fail implicitly and allow some instructions to pass explicitely with `|| true`.